backend-app/models/userModel.js

```diff
@@ -1,55 +1,77 @@
-const mongoose = require("mongoose");
-const validator = require("validator");
-const bcrypt = require("bcryptjs");
+const mongoose = require('mongoose');
+const validator = require('validator');
+const bcrypt = require('bcryptjs');
+const { Actions } = require('../middlewares/authorization');
 
-const userSchema = new mongoose.Schema({
-  name: {
-    type: String,
-    required: [true, "Please fill your name"],
-  },
-  email: {
-    type: String,
-    required: [true, "Please fill your email"],
-    unique: true,
-    lowercase: true,
-    validate: [validator.isEmail, " Please provide a valid email"],
-  },
-  address: {
-    type: String,
-    trim: true,
-  },
-  password: {
-    type: String,
-    required: [true, "Please fill your password"],
-    minLength: 6,
-    select: false,
-  },
-  passwordConfirm: {
-    type: String,
-    required: [true, "Please fill your password confirm"],
-    validate: {
-      validator: function(el) {
-        return el === this.password;
+const userSchema = new mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: [true, 'Please fill your name'],
+    },
+    email: {
+      type: String,
+      required: [true, 'Please fill your email'],
+      unique: true,
+      lowercase: true,
+      validate: [validator.isEmail, ' Please provide a valid email'],
+    },
+    address: {
+      type: String,
+      trim: true,
+    },
+    password: {
+      type: String,
+      required: [true, 'Please fill your password'],
+      minLength: 6,
+      select: false,
+    },
+    passwordConfirm: {
```

Comment:
What do you think about this password confirm should we keep it, on the model, or leave it as a frontend thing?